### ISSUE TITLE
How do I programmatically deselect the completion minibuffer?

### ISSUE THREAD

**Issue by Overdr0ne** [abo-abo/swiper#2653]
I want to use ivy to display information about files in dired. I have ivy-read using an alist of file properties, and the user can select one if they like to yank that property. All that works really nicely.

The problem is, I don't want the information window selected initially. So I've been trying to figure out how to deselect that minibuffer window after completion has started. I tried putting `(select-window (minibuffer-selected-window))` after ivy-read, but that doesn't work it seems because ivy-read hasn't returned until the user selects something. Then I tried creating a custom display function, something like this:
```
(defun ivy-display-function-window (text)
  (let ()
    (insert text)
    (select-window (get-mru-window))
    )
  )
```
But that doesn't seem to work either... Is this possible?

cheers,
Sam

**Comment by basil-conto** [comment]
Have you tried `minibuffer-with-setup-hook`? See [`(info "(elisp) Minibuffer Misc")](https://www.gnu.org/software/emacs/manual/html_node/elisp/Minibuffer-Misc.html).

**Comment by Overdr0ne** [comment]
> Have you tried `minibuffer-with-setup-hook`? See [`(info "(elisp) Minibuffer Misc")`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Minibuffer-Misc.html).

Yeah, that's what I ended up using. I had to customize a lot of other things to get the desired effect, so ivy didn't make a lot of sense anyway. Particularly, fyi, cancelling the previous minibuffer before starting the next to prevent accumulation of minibuffers ended up being really nasty: https://emacs.stackexchange.com/a/20980/29416

**Comment by basil-conto** [comment]
I'm glad you got there in the end.

Does anything remain to be resolved, or can this issue be closed?

**Comment by abo-abo** [comment]
@Overdr0ne Is your code published somewhere? I could have a look to see if the API use or even the API can be improved.

**Comment by Overdr0ne** [comment]
> @Overdr0ne Is your code published somewhere? I could have a look to see if the API use or even the API can be improved.

Yeah, here's where I was trying to use it, but I don't have the hooks in there right now.
https://github.com/Overdr0ne/sfs/blob/master/sfs-recoll.el#L115

The closest I could get was adding to the ivy-hooks-alist:
(defun sfs-ivy-hook ()
  (select-window (get-mru-window)))

(add-to-list 'ivy-hooks-alist
             '(sfs-recoll-dired-ivy . sfs-ivy-hook))

That focused the previous window correctly, but the ivy minibuffer wouldn't populate until it was focused again, and I never figured out why.

**Comment by basil-conto** [comment]
Sounds like there isn't much for Ivy to do here, so closing for now.
